contracts/schemes/ContinuousLockingToken4Reputation.sol

```diff
@@ -182,13 +186,13 @@ contract ContinuousLocking4Reputation is Agreement {
         address(token).safeTransferFrom(msg.sender, address(this), _amount);
         // solhint-disable-next-line not-rely-on-time
         uint256 batchIndexToLockIn = (now - startTime) / batchTime;
+        // [the condition says that _batchIndexToLockIn must be the one corresponding to the "current one" -- so the error message is not very helpful at all]
         require(batchIndexToLockIn == _batchIndexToLockIn, "locking is not active");
-        uint256 j = _period;
-        //fill in the next batches scores.
-        for (int256 i = int256(batchIndexToLockIn + _period - 1); i >= int256(batchIndexToLockIn); i--) {
-            Batch storage batch = batches[uint256(i)];
-            uint256 score = (_period - j + 1) * _amount;
-            j--;
+
+        // fill in the next batches scores.
+        for (int256 p = _period); p > 0; p--) {
+            Batch storage batch = batches[uint256(batchIndexToLockIn + p - 1)];
+            uint256 score = p * _amount;
```

Comment:
this will not give the requested score.
as for locking for n batches : 
S(0) = amount * n 
S(I) = amount * (n - I)  
S(n-1) = amount * 1

Reply:
we can do
for (uint256 p = 0; p < _period; p++) {
            Batch storage batch = batches[batchIndexToLockIn + p];
            uint256 score = ( _period - p) * _amount;